### ISSUE TITLE
 Maximum number of IMAP connections exceeded

### ISSUE THREAD

**Issue by blakegonzales** [joeyates/imap-backup#123]
I've successfully been using imap-backup for a couple years now. Thank you!

Today I ran across this issue in the log after traversing about 75% of my imap folders on the server.

> S: RUBY0001 NO [UNAVAILABLE] Maximum number of connections from user+IP exceeded (mail_max_userip_connections=30)

Is there a way to limit the number of connections imap-backup creates as it runs?

Thanks,
Blake

**Comment by joeyates** [comment]
Hi @blakegonzales thanks for opening the issue.

I've created a fix that should reduce the number of connections to the server.

If you can try it out, please do the following:

```sh
gem uninstall imap-backup # To uninstall the current version
git clone https://github.com/joeyates/imap-backup.git
cd imap-backup/
git checkout --track -b feature/avoid-unnecessary-reconnections origin/feature/avoid-unnecessary-reconnections
gem build imap-backup.gemspec
sudo gem install imap-backup-*.gem
```

**Comment by blakegonzales** [comment]
That worked!  I appreciate your support here :)

**Comment by joeyates** [comment]
This fix is in the 6.0.1 release.
